src/command.c

```diff
@@ -679,8 +687,9 @@ static bool cmd_heapinfo(target_s *t, int argc, const char **argv)
 		target_addr_t heap_limit = strtoul(argv[2], NULL, 16);
 		target_addr_t stack_base = strtoul(argv[3], NULL, 16);
 		target_addr_t stack_limit = strtoul(argv[4], NULL, 16);
-		gdb_outf("heapinfo heap_base: %p heap_limit: %p stack_base: %p stack_limit: %p\n", heap_base, heap_limit,
-			stack_base, stack_limit);
+		gdb_outf("heap_base: %08" PRIx32 " heap_limit: %08" PRIx32 " stack_base: %08" PRIx32 " stack_limit: %08" PRIx32
+				 "\n",
+			heap_base, heap_limit, stack_base, stack_limit);
 		target_set_heapinfo(t, heap_base, heap_limit, stack_base, stack_limit);
 	} else
 		gdb_outf("heapinfo heap_base heap_limit stack_base stack_limit\n");
```

Comment:
```suggestion
		gdb_outf("%s\n", "Set semihosting heapinfo: HEAP_BASE HEAP_LIMIT STACK_BASE STACK_LIMIT");
```
Such that we reuse the *Usage*/help message but still can stick a newline after it. Maybe even `"Usage: %s\n"`.

Reply:
> Such that we reuse the _Usage_/help message but still can stick a newline after it.

That's a good one. Maybe even pass the help string to the command?

```
--- a/src/command.c
+++ b/src/command.c
@@ -155,7 +155,7 @@ int command_process(target_s *const t, char *const cmd_buffer)
                 * So 'mon ver' will match 'monitor version'
                 */
                if ((argc == 0) || !strncmp(argv[0], cmd->cmd, strlen(argv[0])))
-                       return !cmd->handler(t, argc, argv);
+                       return !cmd->handler(t, argc, argv, cmd->help);
        }
 
        if (!t)
```